runtime/vm/resolvesupport.cpp

```diff
@@ -1979,6 +2100,59 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveMethodHandle(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	J9JavaVM *vm = vmThread->javaVM;
+	j9object_t memberName = NULL;
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	bool canRunJavaCode = J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_JIT_COMPILE_TIME | J9_RESOLVE_FLAG_REDEFINE_CLASS);
+
+	if (!canRunJavaCode) {
+		return NULL;
+	}
+
+	J9RAMMethodRef *ramCPEntry = (J9RAMMethodRef*)ramCP + cpIndex;
+	UDATA invokeCacheIndex = ramCPEntry->methodIndexAndArgCount >> 8;
+	J9Class *ramClass = J9_CLASS_FROM_CP(ramCP);
+	J9ROMMethodRef *romMethodRef = (J9ROMMethodRef *)&ramCP->romConstantPool[cpIndex];
+	J9ROMNameAndSignature *nameAndSig = J9ROMMETHODREF_NAMEANDSIGNATURE(romMethodRef);
+
+	/* Resolve the class. */
+	J9Class *resolvedClass = resolveClassRef(vmThread, ramCP, romMethodRef->classRefCPIndex, resolveFlags);
+	if (resolvedClass != NULL) {
+		/* Create appendix array*/
+		J9Class *objectClass = J9VMJAVALANGOBJECT(vm);
+		J9Class *arrayClass = objectClass->arrayClass;
+		if (NULL == arrayClass) {
+			/* Allocate an array class */
+			J9ROMArrayClass* arrayOfObjectsROMClass = (J9ROMArrayClass*)J9ROMIMAGEHEADER_FIRSTCLASS(vmThread->javaVM->arrayROMClasses);
+			arrayClass = vmThread->javaVM->internalVMFunctions->internalCreateArrayClass(vmThread, arrayOfObjectsROMClass, objectClass);
```

Comment:
We're in the VM here and don't need to go through the function table
```suggestion
			arrayClass = internalCreateArrayClass(vmThread, arrayOfObjectsROMClass, objectClass);
```